### DISCUSSION TITLE
Exit on success

### DISCUSSION THREAD

**Discussion by ivanovaleksandar** [community/community#26885] [Actions]
<p>Hi,</p>
<p>I am interested if there is an option to exit the workflow prematurely after a successful step. Is there another status code that can be used for these cases?</p>
<p>For example, if I explicitly return with <code>exit 0</code>, the step will be successful, but the job will continue with the other steps? Is there a way to skip all other steps without using conditionals for every subsequent step.</p>
<pre><code class="lang-auto">      - name: step one
        run: exit 0
      - name: step two
        run: ./something.sh

</code></pre>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/ivanovaleksandar">@ivanovaleksandar</a> ,</p>
<blockquote>
<p>Is there a way to skip all other steps without using conditionals for every subsequent step.</p>
</blockquote>
<p>We have no methods to skip the subsequent steps without using the <strong>if</strong> conditional.<br>
The <strong>if</strong> conditional is the designed and only way that used to skip jobs or steps in the workflow.</p>
<p>In a job, by default only when a step is failure, the subsequent steps will be automatically skipped. Every step in the job will be implicitly added the if statement with the success status (<strong>if: ${{ success() }}</strong>) by default.<br>
If you want to skip the subsequent steps when the previous step is success, you must use the explicit <strong>if</strong> conditional with other statuses to override the default status, or combine with other statements.</p>
<p>Examples:</p>
<ul>
<li>No explicit <strong>if</strong> conditional</li>
</ul>
<pre><code class="lang-yaml">- name: test step
  run: echo "This step does not have the explicit if statement".
# In fact, this step will follow the if statement "if: ${{ success() }}" by default.
# It only runs if all the previous steps in the same job are success.
</code></pre>
<ul>
<li>Override the default status</li>
</ul>
<pre><code class="lang-yaml">- name: test step
  if: ${{ failure() }}
  run: echo "This step does not have the explicit if statement".
# This step will follow the if statement "if: ${{ failure() }}".
# It will run if any previous step in the same job is failure.
</code></pre>
<ul>
<li>Combine with other statements</li>
</ul>
<pre><code class="lang-yaml">- name: test step
  if: ${{ github.event_name == 'push' }}
  run: echo "This step does not have the explicit if statement".
# In fact, this step will follow the if statement
# "if: ${{ success() &amp;&amp; github.event_name == 'push' }}".
# It only runs when all the previous steps in the same job are 
# success AND the event name is 'push'.
</code></pre>

**Comment by ivanovaleksandar** [comment]
<p><a class="mention" href="/brightran">@brightran</a></p>
<p>Thank you for the explanation. After testing this I came to the same conclusion.</p>
<p>Nevertheless, is there a plan for this to change? Because this way, if you have 15 steps afterwards, there is a lot of boilerplate included just for this condition to be fulfilled. Also, it is kind of weird to have support for failure exit code (i.e <code>exit 1</code> will terminate the workflow),  while success means continue (as already mentioned in your answer).</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/ivanovaleksandar">@ivanovaleksandar</a> ,</p>
<blockquote>
<p>it is kind of weird to have support for failure exit code (i.e  <code>exit 1</code>  will terminate the workflow)</p>
</blockquote>
<p>Essentially, it is not that the failed step directly terminate the job or the workflow run. It just follows the implicit statement “<strong>if: ${{ success() }}</strong>” to orderly skip all the subsequent steps to make the job completed. This is consistent with what I mentioned above.</p>
<blockquote>
<p>Nevertheless, is there a plan for this to change?</p>
</blockquote>
<p>I do not find any plan about this.<br>
If you really need the related features, I recommend that you can directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="noopener nofollow ugc">here</a>.<br>
That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>
<p>In addition, you also can follow the “<a href="https://github.com/github/roadmap" rel="noopener nofollow ugc"><strong>GitHub public roadmap</strong></a>” repository where you can learn about what features the appropriate engineering teams are working on, what stage the features are in, and when the teams expect to bring them to you.<br>
When the new features are published, you can view the release note from “<a href="https://github.blog/changelog/" rel="noopener nofollow ugc"><strong>GitHub Changelog</strong></a>”.</p>

**Comment by ivanovaleksandar** [comment]
<p><a class="mention" href="/brightran">@brightran</a></p>
<p>Thank you a lot for the detailed explanation. :+1: :metal:</p>

**Comment by silverqx** [comment]
Has anything changed about this? I'm searching for how to successfully exit the workflow in the middle to skip the rest of the steps.

**Comment by ArkadiuszMichalski** [comment]
I have the same problem, I am unable to terminate entire workflow as success without adding conditions in subsequent steps, which significantly obfuscates the entire code.
